File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.java
import java.io.*;
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        
        int ansX = 0;
        int ansY = 0;
        
        // 하나씩 받아오기
        for (int[] card : sizes) {
            // 두 축 중에서 긴 축을 가로로 지정
            int y = Math.min(card[0] , card[1]);
            int x = Math.max(card[0], card[1]);
            
            // 카드 중 가로/세로가 가장 긴 값 갱신
            ansX = Math.max(ansX, x);
            ansY = Math.max(ansY, y);
        }
        
        answer = ansX * ansY;

        return answer;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    var answer = 0;
    
    var w = 0; // 카드 중 긴 부위를 w로 하겠음
    var h = 0;
    
    for (var i = 0; i < sizes.length; i++) {
        var nowW = sizes[i][0];
        var nowH = sizes[i][1];
        
        // console.log(`${nowW} ${nowH}`);
        
        if (nowW > nowH) {
            var temp = nowW;
            nowW = nowH;
            nowH = temp;
        }
        
        w = Math.max(nowW, w);
        h = Math.max(nowH, h);
        
    }
    
    answer = w * h;
    
    return answer;
}